src/projects/projects-service.js

```diff
@@ -1,8 +1,33 @@
 const xss = require('xss');
 
+const REGEX_ALPHA_NUMBERS_AND_HYPHENS_UNDERSCORE_SPACE = /^[\w]+([-_\s]{1}[a-z0-9]+)*$/i;
 const REGEX_URL_SIMPLE = /^(?:http(s)?:\/\/)?[\w.-]+(?:\.[\w.-]+)+[\w\-._~:/?#[\]@!$&'*+,;=.]+$/;
 
 const ProjectsService = {
+  validateName(name) {
+    name = name.toString();
+    if (name.length < 2) {
+      return 'must be 2 or more characters';
+    }
+
+    if (name.length > 30) {
+      return 'must be less than 30 characters';
+    }
+
+    if (!REGEX_ALPHA_NUMBERS_AND_HYPHENS_UNDERSCORE_SPACE.test(name)) {
+      return 'must contain only alphabetic characters or numbers and only 1 hyphen, underscore or space between them';
+    }
+  },
+  validateProjectDescription(desc) {
+    desc = desc.toString();
+    if (desc.length < 10) {
+      return 'must be 10 or more characters';
+    }
+
+    if (desc.length > 255) {
+      return 'must be less than 255 characters';
```

Comment:
WIth the logic `> 255` let's change the verbiage here a little bit. 

```suggestion
      return 'must be 255 characters or less';
```